data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/SimpleDataService.java

```diff
@@ -91,10 +84,59 @@ public SimpleDataService reader(final DataReader reader) {
         return this;
     }
 
-    public SimpleDataService cacheService(final CacheService cacheService) {
-        requireNonNull(cacheService, "The cache service cannot be set to null.");
-        this.cacheService = cacheService;
-        return this;
+    DataReaderRequest constructReaderRequest(final ReadRequest request) {
+        final GetDataRequestConfig getConfig = new GetDataRequestConfig()
+                .token(new RequestId().id(request.getToken()))
+                .resource(request.getResource());
+        getConfig.setOriginalRequestId(request.getOriginalRequestId());
+        LOGGER.info("Calling palisade service with: {}", getConfig);
+
+        final DataRequestConfig config = getPalisadeService().getDataRequestConfig(getConfig).join();
+        LOGGER.info("Palisade service returned: {}", config);
+
+        final DataReaderRequest readerRequest = new DataReaderRequest()
+                .resource(request.getResource())
+                .user(config.getUser())
+                .context(config.getContext())
+                .rules(config.getRules().get(request.getResource()));
+        readerRequest.setOriginalRequestId(request.getOriginalRequestId());
+
+        return readerRequest;
+    }
+
+    @Override
+    public Consumer<OutputStream> read(final ReadRequest dataRequest) throws NoCapacityException {
+        return out -> {
+            try {
```

Comment:
Can this try block be replaced with a "try with resources" style? e.g. https://www.baeldung.com/java-try-with-resources